Guide/DynamicProgramming/Abbreviation/README.md

```diff
@@ -0,0 +1,115 @@
+# Abbreviation
+
+## Problem
+This problem comes from [HackerRank](https://www.hackerrank.com/contests/world-codesprint-6/challenges/abbr).
+Category: Dynamic Programming
+Difficulty: Medium
+
+### Overview
+Read the link instructions,
+they are helpful when solving
+greater than eye deep
+
+Edge cases and complexities are covered below. Please read the problem first.
+
+## Algorithm
+### Overview
+This problem is complex because of the edge cases that exist. Some
+solutions don't need to worry about them, however when trying to maximize
+efficiency it becomes a problem. When I approached the problem I tried to
+use a double pointer method, iterating through both strings at the same time.
+
+The following example will show why this problem is tough to get right
+the first time, or second.
+
+It looks like this (credit to Orion for the edge case),
+- A = abcBBcA and B = ABBCA
+  - p1 (pointer one) = a | p2 (pointer two) = A
+    - a can be turned into A, so B is possible so far.
+  - p1 = b | p2 = B
+    - Another match
+  - p1 = c | p2 = B
+    - Move p1 forward (since c can be ignored and deleted).
+  - p1 = B | p2 = B
+    - Both are uppercase and they match, so move both pointers forward.
+  - p1 = B | p2 = C
+    - Oops! This situation causes a return of "NO", even though it should
+    print "YES". The problem is that the lowercase 'b' "uses" a slot that
+    should be used by the uppercase 'B'. Because lowercase characters can
+    be deleted but uppercase cannot, **every uppercase character must be
+    used by the algorithm.**
+
+So what's the solution? Well, either you can backtrack whenever you reach
```

Comment:
Do we want to have alternative ideas discussed in the overview of the implemented algorithm? I feel like it might be better to first just go through what works and is implemented in the solution. And then cover alternative solutions in the Conclusion section.